python/rmm/rmm.py

```diff
@@ -267,6 +267,31 @@ def get_ipc_handle(ary, stream=0):
     )
 
 
+def make_cupy_use_rmm():
```

Comment:
Instead of wrapping this in a function like this, what feels cleaner to me is just always creating an RMMCupyMemory class and an `rmm_cupy_allocator` function. Then if we're leaving it as opt-in as opposed to implicitly handling, the user does `cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)`.

Reply:
Yes! 

Also, probably need to test this...  But that would require making cupy a dependency?

Reply:
What about something a bit more generic like `enable_allocator_backend(**kwargs)`, which one could call like this `enable_allocator_backend(cupy=True)`? This would keep the door open for allowing similar functionality for other libraries (much like [`joblib`'s backends]( https://joblib.readthedocs.io/en/latest/parallel.html#custom-backend-api-experimental )). We could even allow this to be plugin friendly down the road to allow others to handle this as they needed. Further down the road other libraries could add themselves without us holding their specific wrapper code.

Reply:
I like both ideas. 
If we go with @kkraus14 idea, we can avoid depending on cupy by implementing an alternative version of `rmm_cupy_allocator` that raises an exception instead, if cupy isn't available.

Reply:
Or if we can accept I minor overhead, we can declare `RMMMemory` instead `rmm_cupy_allocator`.

Reply:
@kkraus14, thoughts? 😉